ADAL/src/telemetry/ADTelemetry.m

```diff
@@ -107,39 +107,45 @@ - (NSString*)registerNewRequest
 - (void)startEvent:(NSString*)requestId
          eventName:(NSString*)eventName
 {
-    if ([NSString adIsStringNilOrBlank:requestId] || [NSString adIsStringNilOrBlank:eventName])
+    @synchronized(self)
     {
-        return;
+        if ([NSString adIsStringNilOrBlank:requestId] || [NSString adIsStringNilOrBlank:eventName])
+        {
+            return;
+        }
+        
+        NSDate* currentTime = [NSDate date];
+        [_eventTracking setObject:currentTime
```

Comment:
The setObject call is the only thing that needs to be in the synchronized block. Try to keep your critical sections as small as possible